lib/routes/builds.js

```diff
@@ -392,8 +430,11 @@ module.exports = function (app) {
     await manager.writeToPath(builtAssets, unpackPath);
     const fileInfos = await manager.walkAndGenerateFileInfos(unpackPath);
     await retry.op(retryOpts).async(() => manager.publishWithBffs(publishSpec, fileInfos));
```

Comment:
can we combine to the 2 cases so we don't have the duplicated code? seems like we just need to do some slightly different operations up front and a different way to get the `fileInfos`